proposals/0012-f3-deals-reliability-ux.md

```diff
@@ -0,0 +1,159 @@
+
+## Project definition
+
+#### Brief plan of attack
+
+**UX track**
+
+1. Study the user flow in detail. Inventorise the specific elements that are confusing users.
+2. Initiate discussions about user-friendlier terminology.
+3. Design a new CLI UX, prototype it, iterate.
+4. Implement CLI UX changes (and underlying code changes)
+5. One-stop set of JSON-RPC operations that encapsulate the new behaviour with no frills (without leaking implementation details as much as possible).
+
+**Technical reliability track**
+
+0. Do a full inventory of outstanding issues; get to the bottom of the long-standing mystical issues; analyse; fix. Some known tasks:
+1. Testing: Design and implement test suites to reproduce the concrete issues and serve as regression tests. Nightly tests for deals.
+2. Fix memory footprint issues.
+3. Fix resumability of data transfers.
+4. Consider p2p file-oriented transfer protocol to cut out complexity for now.
+5. Identify areas of the deals flow code that need more informative logging (eg data transfer restarts / connection retries).
+6. Add logs and journal events.
+7. Fix miner datastore bloat.
+
+#### What does done look like?
+
+_What specific deliverables should completed to consider this project done?_
+
+1. A new version of Lotus (and associated modules) that achieves the success criteria.
+2. A new interactive CLI command (`lotus store`) that guides the user through the process of performing a storage deal, and provides visibility and a polished flow.
+3. A simplified, porcelain S3-like JSON-RPC API that covers the deals flow and provides the logical sequence of steps outlined in previous sections, under a refined, well-documented, and opaque API that doesn't leak Filecoin implementation details.
+4. (Optionally) A prototype for a deals dashboard web UI (in collaboration with The Spark).
+
+#### What does success look like?
+
+_Success means impact. How will we know we did the right thing?_
+
+- 80% of deals initiated via the CLI result in success, as long as prerequisites are met (enough balance, synced node, etc.) under a variety of conditions (e.g. connection losses, slow bandwidth, concurrent deals).
+- Most open issues related to deals on the Filecoin issue tracker are closed, and not reopened, and new ones related to deals are not created shortly thereafter (i.e. we don't introduce new problems).
+- Users give consistent positive feedback about the usability and visibility characteristics of deal-making.
+
+#### Counterpoints & pre-mortem
+
+_Why might this project be lower impact than expected? How could this project fail to complete, or fail to be successful?_
+
+N/A. There's pretty good consensus that this is a high impact project.
```

Comment:
I think the risk here is scope creep. Motivation is great but it can become a “boil the ocean” project with a million sub-project components. How do we scope this down to be maximally valuable and efficient?